js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillPartitionsRequestedDialog.tsx

```diff
@@ -1,57 +1,91 @@
 import {useMemo} from 'react';
 
-import {BackfillTableFragment} from './types/BackfillTable.types';
+import {gql, useQuery} from '../../apollo-client';
+import {
+  BackfillPartitionsDialogContentQuery,
+  BackfillPartitionsDialogContentQueryVariables,
+} from './types/BackfillPartitionsRequestedDialog.types';
 import {TruncatedTextWithFullTextOnHover} from '../../nav/getLeftNavItemsForOption';
 import {VirtualizedItemListForDialog} from '../../ui/VirtualizedItemListForDialog';
 
 const COLLATOR = new Intl.Collator(navigator.language, {sensitivity: 'base', numeric: true});
+
 interface Props {
-  backfill?: BackfillTableFragment;
+  backfillId?: string;
   onClose: () => void;
 }
-export const BackfillPartitionsRequestedDialog = ({backfill, onClose}: Props) => {
+
+export const BackfillPartitionsRequestedDialog = ({backfillId, onClose}: Props) => {
   return (
     <Dialog
-      isOpen={!!backfill}
+      isOpen={!!backfillId}
       title={
         <span>
           Partitions requested for backfill:{' '}
-          <span style={{fontSize: '16px', fontFamily: FontFamily.monospace}}>{backfill?.id}</span>
+          <span style={{fontSize: '16px', fontFamily: FontFamily.monospace}}>{backfillId}</span>
         </span>
       }
       onClose={onClose}
     >
-      <DialogContent partitionNames={backfill?.partitionNames || []} />
+      <DialogContent backfillId={backfillId} />
       <DialogFooter topBorder>
         <Button onClick={onClose}>Done</Button>
       </DialogFooter>
     </Dialog>
   );
 };
 
-interface DialogContentProps {
-  partitionNames: string[];
-}
-
 // Separate component so that we can delay sorting until render.
-const DialogContent = (props: DialogContentProps) => {
-  const {partitionNames} = props;
+const DialogContent = (props: {backfillId: string | undefined}) => {
+  const queryResult = useQuery<
+    BackfillPartitionsDialogContentQuery,
+    BackfillPartitionsDialogContentQueryVariables
+  >(BACKFILL_PARTTIONS_DIALOG_CONTENT_QUERY, {
+    skip: !props.backfillId,
+    variables: {
+      backfillId: props.backfillId ?? '',
+    },
+  });
+
+  const {data, loading} = queryResult;
 
   const sorted = useMemo(() => {
-    return [...(partitionNames || [])].sort((a, b) => COLLATOR.compare(a, b));
-  }, [partitionNames]);
+    const names =
+      data?.partitionBackfillOrError.__typename === 'PartitionBackfill'
+        ? data.partitionBackfillOrError.partitionNames
+        : null;
+
+    return [...(names || [])].sort((a, b) => COLLATOR.compare(a, b));
+  }, [data]);
 
   return (
     <div style={{height: '340px', overflow: 'hidden'}}>
-      <VirtualizedItemListForDialog
-        items={sorted}
-        renderItem={(partitionName) => (
-          <div key={partitionName}>
-            <TruncatedTextWithFullTextOnHover text={partitionName} />
-          </div>
-        )}
-      />
+      {loading ? (
+        <Box padding={48}>
+          <Spinner purpose="page" />
+        </Box>
```

Comment:
Maybe a `<SpinnerWithText />`?

Reply:
Looks like we use this in TickDetailsDialog so I'll copy that to here so it's identical - I'm sorta torn about it because the text makes the loading spinner off center, but we should def be consistent!